test/unit/gateways/secure_pay_au_test.rb

```diff
@@ -27,6 +28,18 @@ def test_supported_card_types
     assert_equal %i[visa master american_express diners_club jcb], SecurePayAuGateway.supported_cardtypes
   end
 
+  def test_endpoint
+    xml_request = @gateway.send(:build_request, :purchase, Builder::XmlMarkup.new.target!)
```

Comment:
for this api version you can do the similar to what I mentioned in the below comment but in `test_localized_currency` for example. If we go to the [purchase](lib/active_merchant/billing/gateways/secure_pay_au.rb) method in looks like the Periodic API only gets used if the payment method doesn't respond to number.